### PR TITLE
Docu fixup and prevent AssertionError

### PR DESCRIPTION
**Pull request by dnltz**

### CODE REVIEW

File: docs/db_dev_process/newpart.rst [hunk 1]
```diff
@@ -100,50 +97,28 @@ Source this new settings file:
...
-at database/<family>/mapping/, which contain the part information (parts.yaml),
-device to fabric mapping (devices.yaml) or hints about resources
-(resources.yaml).
-
-The first file contains a mapping between a part number and informations
-about the device, package and speed grade used by the fuzzers. The
-complete part number is used as key. Device, package and speedgrade are parts
-of the part numbers.
+The project needs to know which device is now available and which fabric it
+uses. Because same devices share the same fabric, this mapping needs to be
```
Comment (litghost) [inline docs/db_dev_process/newpart.rst:101]: `some devices`, not `same devices`

**Review by litghost — APPROVED** [review]
LGTM, we'll see if the CI is green.  Any idea why the CI passed without the `SKIP_ENV` change?

**Comment by dnltz** [comment]
I think it's because the setting scripts does not handle the return code of the callee.
https://github.com/SymbiFlow/prjxray/blob/086f9a1714e96323947d7c2526f5e4b97063e79f/settings/artix7.sh#L36

On the other hand, the fuzzer will fail if the file does not exists.
```
make[1]: *** [Makefile:153: 073-get_counts/run.xc7a100tfgg676-1.ok] Error 1
Traceback (most recent call last):
  File ".//run_fuzzer.py", line 692, in <module>
    sys.exit(main(sys.argv))
  File ".//run_fuzzer.py", line 452, in main
    environment = get_environment_variables()
  File "/home/daniel/work/symbiflow/prjxray/utils/create_environment.py", line 39, in get_environment_variables
    resources = get_part_resources(res_path, os.environ['XRAY_PART'])
  File "/home/daniel/work/symbiflow/prjxray/prjxray/util.py", line 63, in get_part_resources
    "Mapping file {} does not exists".format(filename)
AssertionError: Mapping file /home/daniel/work/symbiflow/prjxray/settings/artix7/resources.yaml does not exists
Traceback (most recent call last):
  File ".//run_fuzzer.py", line 692, in <module>
Traceback (most recent call last):
  File ".//run_fuzzer.py", line 692, in <module>
    sys.exit(main(sys.argv))
  File ".//run_fuzzer.py", line 452, in main
    sys.exit(main(sys.argv))
    environment = get_environment_variables()
  File "/home/daniel/work/symbiflow/prjxray/utils/create_environment.py", line 39, in get_environment_variables
  File ".//run_fuzzer.py", line 452, in main
    resources = get_part_resources(res_path, os.environ['XRAY_PART'])
  File "/home/daniel/work/symbiflow/prjxray/prjxray/util.py", line 63, in get_part_resources
    "Mapping file {} does not exists".format(filename)
AssertionError: Mapping file /home/daniel/work/symbiflow/prjxray/settings/artix7/resources.yaml does not exists
    environment = get_environment_variables()
  File "/home/daniel/work/symbiflow/prjxray/utils/create_environment.py", line 39, in get_environment_variables
    resources = get_part_resources(res_path, os.environ['XRAY_PART'])
  File "/home/daniel/work/symbiflow/prjxray/prjxray/util.py", line 63, in get_part_resources
    "Mapping file {} does not exists".format(filename)
AssertionError: Mapping file /home/daniel/work/symbiflow/prjxray/settings/artix7/resources.yaml does not exists
Traceback (most recent call last):
  File ".//run_fuzzer.py", line 692, in <module>
    sys.exit(main(sys.argv))
  File ".//run_fuzzer.py", line 452, in main
    environment = get_environment_variables()
  File "/home/daniel/work/symbiflow/prjxray/utils/create_environment.py", line 39, in get_environment_variables
    resources = get_part_resources(res_path, os.environ['XRAY_PART'])
  File "/home/daniel/work/symbiflow/prjxray/prjxray/util.py", line 63, in get_part_resources
    "Mapping file {} does not exists".format(filename)
AssertionError: Mapping file /home/daniel/work/symbiflow/prjxray/settings/artix7/resources.yaml does not exists
make[1]: *** [Makefile:87: 000-init-db/run.xc7a50tfgg484-1.ok] Error 1
make[1]: *** [Makefile:87: 000-init-db/run.xc7a100tfgg676-1.ok] Error 1
make[1]: Leaving directory '/home/daniel/work/symbiflow/prjxray/fuzzers'
make[1]: *** [Makefile:155: 075-pins/run.xc7a50tfgg484-1.ok] Error 1
make[1]: Leaving directory '/home/daniel/work/symbiflow/prjxray/fuzzers'
make: *** [Makefile:181: db-artix7] Error 2
make: *** Waiting for unfinished jobs....
make: *** [Makefile:204: db-part-only-artix7_50t] Error 2
make[1]: *** [Makefile:152: 072-ordered_wires/run.xc7a200tffg1156-1.ok] Error 1
make[1]: Leaving directory '/home/daniel/work/symbiflow/prjxray/fuzzers'
make: *** [Makefile:204: db-part-only-artix7_200t] Error 2

```

File: settings/artix7.sh [hunk 2]
```diff
@@ -33,4 +33,9 @@ export XRAY_ROI_GRID_Y2="51"
 
 source $(dirname ${BASH_SOURCE[0]})/../utils/environment.sh
 
-eval $(python3 ${XRAY_UTILS_DIR}/create_environment.py)
+env=`python3 ${XRAY_UTILS_DIR}/create_environment.py`
+ENV_RET=$?
+if [[ $ENV_RET != 0 ]] ; then
+	return $ENV_RET
+fi
+eval $env
```
Comment (dnltz) [inline settings/artix7.sh:41]: @litghost - I have added this check to handle the return code correctly. What do you think?

**Review by litghost — CHANGES_REQUESTED** [review]
Can you rebase this PR and see if it is green now?

**Comment by dnltz** [comment]
@litghost - Rebased and looks good. What do you think?

Maybe we should collect in an issue common failures. The znyq hclk and this kintex seems a little bit unstable righti now.

**Comment by litghost** [comment]
> Maybe we should collect in an issue common failures. The znyq hclk and this kintex seems a little bit unstable righti now.

I agree.  These types of issues arise periodically and get squashed when people drill down and fix them.  If you want to take a look, we always appreciate the help!

File: docs/db_dev_process/newpart.rst [hunk 3]
```diff
@@ -100,50 +97,28 @@ Source this new settings file:
...
-device to fabric mapping (devices.yaml) or hints about resources
-(resources.yaml).
-
-The first file contains a mapping between a part number and informations
-about the device, package and speed grade used by the fuzzers. The
-complete part number is used as key. Device, package and speedgrade are parts
-of the part numbers.
+The project needs to know which device is now available and which fabric it
+uses. Because some devices share the same fabric, this mapping needs to be
+done manuelly. Edit the device.yaml file for the used family under
```
Comment (mithro) [inline docs/db_dev_process/newpart.rst:102]: `manuelly` --> `manually`

File: settings/artix7_200t.sh [hunk 4]
```diff
@@ -23,4 +23,9 @@ export XRAY_IOI3_TILES="RIOI3_X105Y9 LIOI3_X0Y9"
 
 source $(dirname ${BASH_SOURCE[0]})/../utils/environment.sh
 
-eval $(python3 ${XRAY_UTILS_DIR}/create_environment.py)
+env=`python3 ${XRAY_UTILS_DIR}/create_environment.py`
```
Comment (mithro) [inline settings/artix7_200t.sh:26]: Don't use backticks.

**Pull request merged**